File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> ht;
        map<int,int>::iterator it;
        for (int i = 0; i < nums.size(); i++){
            it = ht.find(target - nums[i]);
            if (it == ht.end()){
                ht.insert(pair<int,int>(nums[i],i));
            } else {
                return {i,it->second};
            }
        }
        return {0,0};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = {};
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            if (map.getOrDefault(target - nums[i], -1) != -1){
                ans = new int[]{i, map.get(target - nums[i])};
                return ans;
            } else {
                map.put(nums[i], i);
            }
        }

        return ans;
    }
}